lib/rules/func-names.js

```diff
@@ -33,16 +33,41 @@ module.exports = {
             url: "https://eslint.org/docs/rules/func-names"
         },
 
-        schema: [
-            {
-                enum: ["always", "as-needed", "never"]
-            }
-        ]
+        schema: {
+            definitions: {
+                value: {
+                    enum: [
+                        "always",
+                        "as-needed",
+                        "never"
+                    ]
+                }
+            },
+            items: [
+                {
+                    $ref: "#/definitions/value"
+                },
+                {
+                    type: "object",
+                    properties: {
+                        generators: {
+                            $ref: "#/definitions/value"
+                        }
+                    },
+                    additionalProperties: false
+                }
+            ]
+        }
     },
 
     create(context) {
         const never = context.options[0] === "never";
         const asNeeded = context.options[0] === "as-needed";
+        let requireNamedGenerators = !never;
+
+        if (context.options[1] && context.options[1].hasOwnProperty("generators")) {
```

Comment:
Not sure if this is going to pass linting as is-- I think we now require `Object.prototype.hasOwnProperty.call`.